autogluon/task/image_classification/image_classification.py

```diff
@@ -268,30 +270,15 @@ def fit(dataset,
             lr_config=lr_config
         )
 
-        scheduler_options = {
-            'resource': {'num_cpus': nthreads_per_trial, 'num_gpus': ngpus_per_trial},
-            'checkpoint': checkpoint,
-            'num_trials': num_trials,
-            'time_out': time_limits,
-            'resume': resume,
-            'visualizer': visualizer,
-            'time_attr': 'epoch',
-            'reward_attr': 'classification_reward',
-            'dist_ip_addrs': dist_ip_addrs,
-            'searcher': search_strategy,
-            'search_options': search_options,
-            'plot_results': plot_results
-        }
-        if search_strategy == 'hyperband':
-            scheduler_options.update(
-                {
-                    'searcher': 'random',
-                    'max_t': epochs,
-                    'grace_period': grace_period if grace_period else epochs // 4
-                }
-            )
-
-        results = BaseTask.run_fit(train_image_classification, search_strategy, scheduler_options)
+        scheduler_options = compile_scheduler_options(
```

Comment:
keyword arguments should be used with this many inputs, it is very hard to notice if an arg is in the wrong position currently.

Reply:
OK